src/Microsoft.DotNet.Interactive/Parsing/PolyglotLexer.cs

```diff
@@ -56,11 +56,11 @@ private void LexTrivia()
             }
         }
 
-        private void LexDirective()
+        private bool IsDirective()
```

Comment:
Why the name change? This differs from the pattern, and perhaps obscures that fact that this is also triggering a state change.